ludwig/models/trainer.py

```diff
@@ -1144,7 +1144,10 @@ def check_progress_on_validation(
                 )
 
         # ========== Early Stop logic ==========
-        if 0 < early_stop <= progress_tracker.last_improvement:
+        should_early_stop = torch.Tensor([0 < early_stop <= progress_tracker.last_improvement])
```

Comment:
Might want to use an int type tensor here just to be safe:

```
torch.as_tensor([...], dtype=torch.int)
```

Reply:
Done